app/src/main/java/com/jetbrains/handson/mpp/mobile/AdvancedSearchActivity.kt

```diff
@@ -0,0 +1,148 @@
+import android.content.Intent
+import android.text.format.DateFormat
+import android.widget.DatePicker
+import android.widget.TextView
+import android.widget.TimePicker
+import kotlinx.android.synthetic.main.activity_pop_up_window.*
+import java.util.*
+
+class AdvancedSearchActivity : AppCompatActivity(), AdvancedSearchContract.View, DatePickerDialog.OnDateSetListener,
+        TimePickerDialog.OnTimeSetListener {
+
+        private lateinit var presenter: AdvancedSearchPresenter
+
+        var day = 0
+        var month: Int = 0
+        var year: Int = 0
+        var hour: Int = 0
+        var minute: Int = 0
+        var timeChanged = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_pop_up_window)
+
+        presenter = AdvancedSearchPresenter()
+        presenter.onViewTaken(this)
+
+        setInitialValues()
+        setListeners()
+
+    }
+
+    private fun setInitialValues() {
+        val numAdultsView = findViewById<TextView>(R.id.in_num_adults)
+        numAdultsView.text = "1"
+        val numChildrenView = findViewById<TextView>(R.id.in_num_children)
+        numChildrenView.text = "0"
+        val timeView = findViewById<TextView>(R.id.in_date)
+        timeView.text = "Today, Now"
+    }
+
+    override fun showAlertMessage(alertMessage: String) {
+        val alertDialogBuilder = AlertDialog.Builder(this)
+        alertDialogBuilder.setMessage(alertMessage)
+        alertDialogBuilder.setNeutralButton("Ok") { _, _ -> null}
+        alertDialogBuilder.show()
+    }
+
+    private fun setListeners() {
+        val exitPopUpButton = findViewById<Button>(R.id.submit_advanced_search)
+        exitPopUpButton.setOnClickListener {
+            presenter.submitSearch(getNumAdults(), getNumChildren(), getDateInAPIFormat())
+        }
+
+        val dateButton = findViewById<Button>(R.id.btn_date)
+        dateButton.setOnClickListener {
+            showDatePicker()
+        }
+
+        val numAdultsView = findViewById<TextView>(R.id.in_num_adults)
+        val numChildrenView = findViewById<TextView>(R.id.in_num_children)
+        add_num_adults.setOnClickListener {
+            numAdultsView.text = stepValueXbyY(getNumAdults(),1).toString()
+        }
+        add_num_children.setOnClickListener {
+            numChildrenView.text = stepValueXbyY(getNumChildren(),1).toString()
+        }
+        minus_num_adults.setOnClickListener {
+            numAdultsView.text = stepValueXbyY(getNumAdults(),-1).toString()
+        }
+        minus_num_children.setOnClickListener {
+            numChildrenView.text = stepValueXbyY(getNumChildren(),-1).toString()
+        }
+    }
+    override fun submitAdvancedSearch() {
+        val intent = Intent(this, MainActivity::class.java)
+        intent.putExtra("numAdults", getNumAdults() )
+        intent.putExtra("numChildren", getNumChildren())
+        if (timeChanged){
```

Comment:
Something that could be useful (not necessarily exactly here) is that you can initialise a DateTime object from the year/month/day etc by doing 
```
val datetime = DateTime(year, month, day, hour, minute)
```
which might make it easier for you to pass around a DateTime that's either now or constructed from the selected time, and then just use a DateTimeFormat to produce the correctly formatted string for the api

Reply:
If you want to include klock in the app code so you can use DateTime and you haven't already, you need to add 
```
implementation("com.soywiz.korlibs.klock:klock:1.10.0")
```
to the app `build.gradle` file